lambdas/shared/t4_lambda_shared/preview.py

```diff
@@ -75,20 +76,22 @@ def extract_fcs(file_, as_html=True):
     data = None
     body = ""
     info = {}
-    # per Lambda docs we can use tmp/*, OK to overwrite
-    file_path = os.path.join(TEMP_DIR, "quilt-temp.fcs")
     # fcsparser only takes paths, so we need to write to disk; OK because
-    # FCS files typically ~1MB
-    with open(file_path, 'wb') as real_file:
-        real_file.write(file_.read())
-    try:
-        meta, data = fcsparser.parse(file_path, reformat_meta=True)
-    except Exception as first:  # pylint: disable=broad-except
+    # FCS files typically < 500MB (Lambda disk)
+    # per Lambda docs we can use tmp/*, OK to overwrite
+    with tempfile.NamedTemporaryFile(prefix=TEMP_DIR) as tmp:
+        inbound = file_.read()
```

Comment:
If they can be up to 500MB, this should really de done in a loop. Or maybe `shutil.copyfileobj` will do the right thing.

Reply:
we have 1GB+ in our lambdas but you're right, no reason to be sloppy with memory.